src/ansys/simai/core/data/models.py

```diff
@@ -21,9 +21,85 @@
 # SOFTWARE.
 
 from dataclasses import asdict, dataclass
-from typing import Any, Dict, List
+from typing import Any, Dict, List, Literal
+
+from pydantic import BaseModel, PositiveFloat, ValidationError, model_validator
 
 from ansys.simai.core.data.base import ComputableDataModel, Directory
+from ansys.simai.core.errors import InvalidArguments
+
+
+class DomainAxisDefinition(BaseModel):
+    """Representation of an axis in Domain of Analysis.
+
+    Args:
+        position: indicates the position of the anchor.
+
+                    | *relative_to_min*: it corresponds to `VolXmin = xmin - value`
+
+                    | *relative_to_max*: it corresponds to `VolXmax = xmax + value`
+
+                    | *relative_to_center*: it corresponds to `(xmin+xmax)/2 - value`
+
+                    | *absolute*: it corresponds to `VolXmin = value`
+
+        value:      the distance of the anchor from the position. When position is set to
+                    "absolute", value could be either positive or negative. Only positive
+                    values are accepted in any other case.
+        length:     the length of the Domain of Analysis along the axis.
+
+    """
+
+    position: Literal["relative_to_min", "relative_to_max", "relative_to_center", "absolute"]
+    value: float
+    length: PositiveFloat
+
+    @model_validator(mode="after")
+    def chek_value(self):
+        """Evaluates that the value is set correctly according to the position argument."""
+        if self.value < 0 and self.position != "absolute":
+            raise InvalidArguments(
+                "value should be a positive number when the position is not absolute"
+            )
+        return self
+
+
+@dataclass
+class DomainOfAnalysis:
+    """Definition of the Domain of Analysis.
+
+    Args:
+        Length: the definition of the Domain of Analysis along the X axis
+        Width: the definition of the Domain of Analysis along the Y axis
+        Height: the definition of the Domain of Analysis along the Z axis
+
+    Example:
+        Get the Domain of Analysis from a configuration and replace it with a new one
+
+        .. code-block:: python
+
+            # get the last configuration from a project
+            bld_conf = my_project.last_model_configuration
+
+            # get DoA from the configuration
+            doa = bld_conf.domain_of_analysis
+
+            # create a new DoA
+            new_doa = simai.models.domain_of_analysis()
+
+            new_doa.Height = simai.models.doa_axis_definition("relative_to_min", 4.5, 4.1)
+            new_doa.Length = simai.models.doa_axis_definition("relative_to_center", 0.5, 15)
+            new_doa.Width = simai.models.doa_axis_definition("absolute", -1.5, 6.1)
+
+            # set the new DoA to the configuration
+
+            bld_conf.domain_of_analysis = new_doa
+
+    """
+
+    Length: DomainAxisDefinition = None
+    Width: DomainAxisDefinition = None
+    Height: DomainAxisDefinition = None
```

Comment:
I used upper case for the first letter so that it matches the definition in the UI; given that users would first use the UI and then the SDK, it would be straightforward for them to affiliate the parameters. Also, I wanted to avoid having `doa.length.length`. Anyways, I updated the names of the attributes. Resolving this comment and the other comments related to this one.